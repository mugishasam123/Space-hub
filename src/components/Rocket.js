import React from "react";
import './styles/rocket.css'
const Rocket = ({rocket}) => {
  return (
    <div className="rocket">
      <img src={rocket.image} alt="" className="rocket-image"/>
      <div className="details">
        <h3 className="title">{rocket.name}</h3>
        <p className="description">{rocket.description}</p>
        <button className="reserve-cancel">Reserve Rocket</button>
      </div>
    </div>
  );
};

export default Rocket;
