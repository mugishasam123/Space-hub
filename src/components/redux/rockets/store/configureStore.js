import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { rocketReducer } from '../reducers/rocketReducer';
import { missionReducer } from '../../missions/missionActions';

const rootReducer = combineReducers({
  rockets: rocketReducer,
  missions: missionReducer,
});

const Store = createStore(rootReducer, applyMiddleware(thunk, logger));

export default Store;
