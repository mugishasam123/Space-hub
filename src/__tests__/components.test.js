import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../components/redux/rockets/store/configureStore';
import Rocket from '../components/Rocket';
import RocketsList from '../components/Rockets';

it('Test if Rocket renders correctly', () => {
  const item = {
    description: 'description',
    image: 'https://test.com',
    name: 'Rocket',
    id: 1,
    reserved: false,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Rocket rocket={item} />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test if Rockets List renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <RocketsList />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
