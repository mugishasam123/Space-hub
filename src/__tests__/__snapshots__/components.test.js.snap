// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Rocket renders correctly 1`] = `
<div
  className="rocket"
>
  <img
    alt=""
    className="rocket-image"
    src="https://test.com"
  />
  <div
    className="details"
  >
    <h3
      className="title"
    >
      Rocket
    </h3>
    <p
      className="description"
    >
      <span>
        description
      </span>
    </p>
    <button
      className="reserve-cancel"
      name="reservebtn"
      onClick={[Function]}
      type="button"
    >
      RESERVE ROCKET
    </button>
  </div>
</div>
`;

exports[`Test if Rockets List renders correctly 1`] = `
<section
  className="rockets"
/>
`;
